dotnetv3/EventBridge Scheduler/Scenarios/SchedulerWorkflow.cs

```diff
@@ -0,0 +1,655 @@
+            Console.WriteLine(new string('-', 80));
+
+            if (prepareSuccess)
+            {
+                Console.WriteLine(new string('-', 80));
+                await CreateOneTimeSchedule();
+                Console.WriteLine(new string('-', 80));
+
+                Console.WriteLine(new string('-', 80));
+                await CreateRecurringSchedule();
+                Console.WriteLine(new string('-', 80));
+            }
+
+            Console.WriteLine(new string('-', 80));
+            await Cleanup();
+            Console.WriteLine(new string('-', 80));
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "There was a problem with the workflow, initiating cleanup...");
+            _interactive = false;
+            await Cleanup();
+        }
+
+        Console.WriteLine("Amazon EventBridge Scheduler workflow completed.");
+    }
+
+    /// <summary>
+    /// Prepares the application by creating the necessary resources.
+    /// </summary>
+    /// <returns>True if the application was prepared successfully.</returns>
+    public static async Task<bool> PrepareApplication()
+    {
+        Console.WriteLine("Preparing the application...");
+        try
+        {
+            // Prompt the user for an email address to use for the subscription.
+            Console.WriteLine("\nThis example creates resources in a CloudFormation stack, including an SNS topic" +
+                          "\nthat will be subscribed to the EventBridge Scheduler events. " +
+                          "\n\nYou will need to confirm the subscription in order to receive event emails. ");
+
+            var emailAddress = PromptUserForEmail();
+
+            // Prompt the user for a name for the CloudFormation stack
+            _stackName = PromptUserForStackName();
+
+            // Deploy the CloudFormation stack
+            var deploySuccess = await DeployCloudFormationStack(_stackName, emailAddress);
+
+            if (deploySuccess)
+            {
+                // Create a schedule group for all workflow schedules
+                await _schedulerWrapper.CreateScheduleGroupAsync(_scheduleGroupName);
+
+                Console.WriteLine("Application preparation complete.");
+                return true;
+            }
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while preparing the application.");
+        }
+        Console.WriteLine("Application preparation failed.");
+        return false;
+    }
+
+    /// <summary>
+    /// Deploys the CloudFormation stack with the necessary resources.
+    /// </summary>
+    /// <param name="stackName">The name of the CloudFormation stack.</param>
+    /// <param name="email">The email to use for the subscription.</param>
+    /// <returns>True if the stack was deployed successfully.</returns>
+    private static async Task<bool> DeployCloudFormationStack(string stackName, string email)
```

Comment:
Would it make sense to pull this CFN code into a separate library? It strikes me that you'll likely want to reuse it in many examples.

Reply:
It may make sense in the future, but I prefer to keep these as self-contained as possible. If I end up duplicating it exactly again, I'll re-evaluate.